lib/spack/spack/schema/container.py

```diff
@@ -8,16 +8,18 @@
     'type': 'object',
     'additionalProperties': False,
     'properties': {
-        'os': {
-            'type': 'string',
-            'enum': ['ubuntu:18.04',
-                     'ubuntu:16.04',
-                     'centos:7',
-                     'centos:6']
-        },
-        'spack': {
-            'type': 'string',
-        },
+        'os': {'type': 'string'},
+        'spack': {'anyOf': [
+            {'type': 'string'},
+            {'type': 'object',
+             'additional_properties': False,
+             'properties': {
+                 'url': {'type': 'string'},
+                 'ref': {'type': 'string'},
+                 'enforce-sha': {'type': 'boolean', 'default': False},
```

Comment:
_ or -?

Reply:
I think `_` it's better for consistency with other configuration option. Good catch.